posthog/integrations/django.py

```diff
@@ -0,0 +1,116 @@
+from typing import TYPE_CHECKING, cast
+from posthog import scopes
+
+if TYPE_CHECKING:
+    from django.http import HttpRequest, HttpResponse  # noqa: F401
+    from typing import Callable, Dict, Any, Optional  # noqa: F401
+
+
+class PosthogContextMiddleware:
+    """Middleware to automatically track Django requests.
+
+    This middleware wraps all calls with a posthog scope. It attempts to extract the following from the request headers:
+    - Session ID as $session_id, (extracted from `X-POSTHOG-SESSION-ID`)
+    - Distinct ID as $distinct_id, (extracted from `X-POSTHOG-DISTINCT-ID`)
```

Comment:
Is this set in our SDK? I didn't see it referenced anywhere but maybe it's coming in a separate PR. I know we append `X-POSTHOG-SESSION-ID` when tracing headers are enabled

Reply:
This is all it should be I think: https://github.com/PostHog/posthog-js/pull/2026